lib/decodeLogs.js

```diff
@@ -1,10 +1,17 @@
 const SolidityEvent = require('web3/lib/web3/event.js');
 
 function decodeLogs (logs, contract) {
-  return logs.map(log => {
-    const event = new SolidityEvent(null, contract.events[log.topics[0]], '0x0');
-    return event.decode(log);
-  });
+  const decodedLogs = [];
+  
+  for (const log of logs) {
+    const eventName = log.topics[0];
+    if (eventName in contract.events) {
+      const eventType = new SolidityEvent(null, contract.events[eventName], '0x0');
+      decodedLogs.push(eventType.decode(log));
+    }
+  }
+        
+  return decodedLogs;
```

Comment:
What about doing the ES6 way? it could be sth like:
```js
return logs.reduce((decodedLogs, log) => {
   const eventName = log.topics[0]
   if(!(eventName in contract.events)) return decodedLogs;
   const eventType = new SolidityEvent(null, contract.events[eventName], '0x0');
   return decodedLogs.concat(eventType.decode(log));
}, [])
```

Reply:
I would keep the `for` loop, I find it much easier to read.

Reply:
Good. This basically adds a check for the topic existing in the contract's events before just trying to read it.
Be aware that CLI calls this with:
```
    const logs = decodeLogs([receipt.logs[1]], UpgradeabilityProxyFactory);
```
i.e. with only the 2nd log.
We would have to change usage of decodeLogs in the CLI and elsewhere to:
```
    const logs = decodeLogs(receipt.logs, UpgradeabilityProxyFactory);
```